### ISSUE TITLE
Nested lambda functions with the same parameter names generates an incorrect error

### ISSUE THREAD

**Issue by jamesqo** [dotnet/roslyn#17402]
```cs
using System;
using System.Threading;

public class C {
    public void M()
    {
        ThreadPool.QueueUserWorkItem(foo =>
        {
            ThreadPool.QueueUserWorkItem(foo =>
            {
            });
        });
    }
}
```

This generates the following error at the second call to `QueueUserWorkItem`, saying that it conflicts with another *type* parameter:

```
(13,41,13,44): error CS0412: 'foo': a parameter, local variable, or local function cannot have the same name as a method type parameter
```

[TryRoslyn repro](http://tryroslyn.azurewebsites.net/#b:master/f:>ilr/K4Zwlgdg5gBAygTxAFwKYFsDcAoUlaIoYB0AKgBYBOqAhgCb47YAOwARgDZgDGM3HNECBgBhGAG9sMaTFaceMFDWQKAbgHswdGABkAPKTjrglbqgB8ACgCUUmZJmOYqmpRiHjp1DAC8MZAjMqOoAZpYeJmbWOI4AvnbSCbLsXLwaWjAAsjZJDk7SFNT0AArq6hzEAIrAqDUAqiColADq6pQA1gCSaOiWIWW+5kmOefmOhbR0peVVNfWNLW1dPX0DPkMAkFujY9Kx0cMy+zFH2LFAA===)

**Comment by gafter** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/17248
